solidity/contracts/bridge/WalletCoordinator.sol

```diff
@@ -0,0 +1,542 @@
+    /// @param _depositSweepMaxSize New value.
+    /// @dev Requirements:
+    ///      - The caller must be the owner.
+    function updateDepositSweepMaxSize(uint16 _depositSweepMaxSize)
+        external
+        onlyOwner
+    {
+        depositSweepMaxSize = _depositSweepMaxSize;
+        emit DepositSweepMaxSizeUpdated(_depositSweepMaxSize);
+    }
+
+    /// @notice Submits a deposit sweep proposal. Locks the target wallet
+    ///         for a specific time, equal to the proposal validity period.
+    ///         This function does not store the proposal in the state but
+    ///         just emits an event that serves as a guiding light for wallet
+    ///         off-chain members. Wallet members are supposed to validate
+    ///         the proposal on their own, before taking any action.
+    /// @param proposal The actual deposit sweep proposal
+    /// @param walletMemberContext Optional parameter holding some data allowing
+    ///        to confirm the wallet membership of the caller. This parameter is
+    ///        relevant only when the caller is not a registered proposal
+    ///        submitter but claims to be a member of the target wallet.
+    /// @dev Requirements:
+    ///      - The caller is either a proposal submitter or a member of the
+    ///        target wallet,
+    ///      - The wallet is not time locked.
+    function submitDepositSweepProposal(
+        DepositSweepProposal calldata proposal,
+        WalletMemberContext calldata walletMemberContext
+    )
+        external
+        onlyProposalSubmitterOrWalletMember(
+            proposal.walletPubKeyHash,
+            walletMemberContext
+        )
+        onlyAfterWalletLock(proposal.walletPubKeyHash)
+    {
+        walletLock[proposal.walletPubKeyHash] =
+            /* solhint-disable-next-line not-rely-on-time */
+            uint32(block.timestamp) +
+            depositSweepProposalValidity;
+
+        emit DepositSweepProposalSubmitted(proposal, msg.sender);
+    }
+
+    /// @notice View function encapsulating the main rules of a valid deposit
+    ///         sweep proposal. This function is meant to facilitate the off-chain
+    ///         validation of the incoming proposals. Thanks to it, most
+    ///         of the work can be done using a single readonly contract call.
+    ///         Worth noting, the validation done here is not exhaustive as some
+    ///         conditions may not be verifiable within the on-chain function or
+    ///         checking them may be easier on the off-chain side. For example,
+    ///         this function does not check the SPV proofs and confirmations of
+    ///         the deposit funding transactions as this would require an
+    ///         integration with the difficulty relay that greatly increases
+    ///         complexity. Instead of that, each off-chain wallet member is
+    ///         supposed to do that check on their own.
```

Comment:
For now, it's just the validation mentioned in the example. Not sure if we need something else so I wanted to avoid being too specific.

Reply:
It's also about validating confirmations of deposits revealed to Ethereum. How it's done off-chain is another discussion. It might mean checking if the current`depositMinAge` is enough. I think the client should do something. There are weird scenarios like Ethereum fork with `depositMinAge` lowered to 0 and proposing a sweep of non-confirmed transactions. This is quite a complicated scenario but if we can avoid it with a simple call, I'd do it.

I'd just add something like that at the end of the `@dev Requirements` section:

```
The following off-chain validation must be performed as a bare minimum:
- Inputs used for the sweep transaction have enough Bitcoin confirmations,
- Deposits revealed to the Bridge have enough Ethereum confirmations.
```

Reply:
See 0848de33